### ISSUE TITLE
Magento fails with xdebug 3 due to mftf

### ISSUE THREAD

**Issue by kandy** [magento/magento2-functional-testing-framework#808]
### Preconditions

1. Install xdebug 3 (RC1+) and enable it

### Steps to reproduce

1.  Run bin/magento

### Expected result

1. List of available command is printed

### Actual result

```
PHP Fatal error:  Uncaught Error: Call to undefined function xdebug_disable() in /magento/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php:81
Stack trace:
#0 /magento/vendor/composer/autoload_real.php(73): require()
#1 /magento/vendor/composer/autoload_real.php(63): composerRequire4012c480a3949eb57e5b1ec1b7b5c42b('eda65932675b68b...', '/magento/vendor...')
#2 /magento/vendor/autoload.php(7): ComposerAutoloaderInit4012c480a3949eb57e5b1ec1b7b5c42b::getLoader()
#3 /magento/app/autoload.php(51): include('/magento/vendor...')
#4 /magento/app/bootstrap.php(33): require_once('/magento/app/au...')
#5 /magento/bin/magento(14): require('/magento/app/bo...')
#6 {main}
  thrown in /magento/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php on line 81
```
see for details https://3.xdebug.org/docs/upgrade_guide

**Comment by jilu1** [comment]
@kandy Thank you for your report! 
https://jira.corp.magento.com/browse/MQE-2391 has been created. Please check comments in MQE-2391 and let us know if this needs immediate attention.

**Comment by convenient** [comment]
A bunch of our test suites started failing in travis because of this error.

I think travis has likely started bundling xdebug 3 or something like that.

```yml
before_install:
  - phpenv config-rm xdebug.ini || true
```

I have slipped this into our `.travis.yml` and the suites are running again.

**Comment by jilu1** [comment]
@convenient It's good to know you have a solution!
If your problem is the same as this xdebug_disable() in _bootstrap.php, it's been fixed and available in `develop` branch. No planned release date yet.

**Comment by tuyennn** [comment]
@magento-engcom-team possible to provide the commit related to the fix?

**Comment by densen45** [comment]
@tuyennn I think these are the two commits you asked for: 

e95fb10d3a55d72af5d60e07703d09201812599e
6d6a7c59fc789625345c0b7b071fe08ea68b487b

Please, correct me, if I'm wrong.

**Comment by tuyennn** [comment]
@densen45 @jilu1 hopefully the release could start soon

**Comment by khanhnhat1660407** [comment]
https://xdebug.org/docs/upgrade_guide
![image](https://user-images.githubusercontent.com/43997361/101458244-36487680-3969-11eb-9feb-4e2ea564f286.png)

**Comment by Ddcdom** [comment]
`xdebug_disable` is only called if both MFTF debugging is disabled and the Xdebug module is loaded.
Its possible to enable MFTF debugging by adding/editing the file dev/tests/acceptance/.env and adding the line `MFTF_DEBUG = 1`
I've done this in my dev environment, but as I don't understand the consequences of enabling MFTF debugging I have not enabled it in my production environment. However, as I don't have Xdebug installed in my production environment, this is not an issue.

**Comment by larsroettig** [comment]
@jilu1 @kandy Pls Realse ASAP most Partners use Integration,Static, Mftf Tests on CI-System alle our tests are RED beausce of this issue. Build Patch for us and need to apply them now to all our customers

**Comment by azambon** [comment]
Not only `bin/magento` fails, also web requests crash with xdebug 3. Which means that it's impossibile to debug web requests with xdebug 3

**Comment by larsroettig** [comment]
Yea because most thinks runs over this `_boostrap` in dev mode :sweat_smile:

**Comment by lfluvisotto** [comment]
@magento I am working on this

**Comment by lfluvisotto** [comment]
Replace:

if (!(bool)$debugMode && extension_loaded('xdebug')) {
    xdebug_disable();
}

By:

if (!(bool)$debugMode && extension_loaded('xdebug')) {
    if (function_exists('xdebug_disable')) {
        xdebug_disable();
    }
}

**Comment by jilu1** [comment]
@larsroettig @tuyennn @lfluvisotto  Thanks for the information! We will try to release a patch soon.

**Comment by larsroettig** [comment]
Would be so nice thanks in advance :+1:

**Comment by larsroettig** [comment]
@lfluvisotto you dont need open a PR is already fixed but not released

**Comment by jilu1** [comment]
Here is the fix, please let us know if you see any other issue.
https://github.com/magento/magento2-functional-testing-framework/commit/f99fbbcf19a11822ff16a493dd30fbaf2edeb296

**Comment by larsroettig** [comment]
> Here is the fix, please let us know if you see any other issue.
> [f99fbbc](https://github.com/magento/magento2-functional-testing-framework/commit/f99fbbcf19a11822ff16a493dd30fbaf2edeb296)

This fix works fine for us

**Comment by Lmarcho** [comment]
> Here is the fix, please let us know if you see any other issue.
> [f99fbbc](https://github.com/magento/magento2-functional-testing-framework/commit/f99fbbcf19a11822ff16a493dd30fbaf2edeb296)

after this fix xdebug is not working on that project :disappointed:

**Comment by jilu1** [comment]
@Lmarcho 
Could you provide more details?

The fix is not to disable xdebug and I don't think it will affect how xdebug works.

**Comment by jilu1** [comment]
This is now available in 3.2.1

**Comment by TheRealHunzik** [comment]
Working Solution : 

Open `/etc/php/7.4/mods-available/xdebug.ini` in your favorite Editior

Add These Lines

```
xdebug.mode=debug
xdebug.client_port=9000
```

Your Debugger will start Working
